File: MyControls/gridlayout/src/main/java/zs/xmx/wygridlayout/MainActivity.java
package zs.xmx.wygridlayout;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private List<String>     mTitles;
    private DragedGridLayout mDragedGridLayout;
    private DragedGridLayout mHideGridLayout;
    private List<String>     mNewtitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initData();
        initView();

    }

    private void initView() {
        setContentView(R.layout.activity_main);
        mDragedGridLayout = (DragedGridLayout) findViewById(R.id.dragedgridlayout);

        mDragedGridLayout.setAllowDrag(true);
        mDragedGridLayout.setItem(mTitles);
        mDragedGridLayout.setOnDragItemClickListener(new DragedGridLayout.onDragItemClickListener() {
            @Override
            public void onDragItemClick(TextView textView) {
                //移除view,由于removeView需要时间,因此不能直接addView,否者会报错
                mDragedGridLayout.removeView(textView);
                mHideGridLayout.addItem(textView.getText().toString());
            }
        });

        mHideGridLayout = (DragedGridLayout) findViewById(R.id.hidegridlayout);

        mHideGridLayout.setAllowDrag(false);
        mHideGridLayout.setItem(mNewtitle);
        mHideGridLayout.setOnDragItemClickListener(new DragedGridLayout.onDragItemClickListener() {
            @Override
            public void onDragItemClick(TextView textView) {
                //添加view
                mHideGridLayout.removeView(textView);
                mDragedGridLayout.addItem(textView.getText().toString());
            }
        });
    }

    private void initData() {
        //定义出频道数据
        mTitles = new ArrayList<>();
        mTitles.add("上海");
        mTitles.add("北京");
        mTitles.add("广州");
        mTitles.add("深圳");
        mTitles.add("泉州");
        mTitles.add("漳州");
        mTitles.add("昆明");
        mTitles.add("西安");
        mTitles.add("昆仑");
        mTitles.add("峨眉");
        mTitles.add("神农架");

        //新的频道数据

        mNewtitle = new ArrayList<>();
        mNewtitle.add("中山");
        mNewtitle.add("湛江");
        mNewtitle.add("长江");
        mNewtitle.add("东莞");
        mNewtitle.add("茂名");
        mNewtitle.add("汕头");
        mNewtitle.add("长白山");
        mNewtitle.add("大兴安岭");

    }


}

File: MyControls/app/src/main/java/zs/xmx/mycontrols/MainActivity.kt
package zs.xmx.mycontrols

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import zs.xmx.ExpandEditTextActivity
import zs.xmx.controls.DialogActivity
import zs.xmx.lib_ddmenu.DropDownMenuActivity
import zs.xmx.mycontrols.bottom_sheet.BottomSheetActivity
import zs.xmx.mycontrols.expand_textview.ExpandTextViewActivity
import zs.xmx.mycontrols.shape_shadow.ShapeDrawableActivity
import zs.xmx.progressbar.ProgressBarActivity
import zs.xmx.rvgather.GatherRecycleViewActivity
import zs.xmx.timer.TimerActivity
import zs.xmx.wygridlayout.DragedGridLayoutActivity

class MainActivity : AppCompatActivity(), AdapterView.OnItemClickListener {
    private lateinit var mList: MutableList<String>
    private lateinit var listView: ListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initView()
        initData()
        initEvent()
    }

    private fun initEvent() {
        listView.adapter = ArrayAdapter(this, R.layout.activity_main, mList)
        listView.onItemClickListener = this
    }

    private fun initData() {
        mList = ArrayList()
        mList.add("DialogFragment")
        mList.add("RecycleView")
        mList.add("ExpandEditText")
        mList.add("DragGridLayout")
        mList.add("Timer")
        mList.add("ProgressBar")
        mList.add("ShapeDrawable")
        mList.add("ExpandTextView")
        mList.add("BottomSheet 实现三段式折叠控件")
        mList.add("DropDownMenu 多条件筛选菜单")
    }

    private fun initView() {
        listView = ListView(this)
        setContentView(
            listView, ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup
                    .LayoutParams.MATCH_PARENT
            )
        )

    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        when (position) {
            0 -> startActivity(Intent(this, DialogActivity::class.java))
            1 -> startActivity(Intent(this, GatherRecycleViewActivity::class.java))
            2 -> startActivity(Intent(this, ExpandEditTextActivity::class.java))
            3 -> startActivity(Intent(this, DragedGridLayoutActivity::class.java))
            4 -> startActivity(Intent(this, TimerActivity::class.java))
            5 -> startActivity(Intent(this, ProgressBarActivity::class.java))
            6 -> startActivity(Intent(this, ShapeDrawableActivity::class.java))
            7 -> startActivity(Intent(this, ExpandTextViewActivity::class.java))
            8 -> startActivity(Intent(this, BottomSheetActivity::class.java))
            9 -> startActivity(Intent(this, DropDownMenuActivity::class.java))
        }
    }


}
